plugins/dell-kestrel/fu-dell-kestrel-ec.c

```diff
@@ -0,0 +1,1052 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-kestrel-common.h"
+
+typedef struct __attribute__((packed)) { /* nocheck:blocked */
+	struct FuDellKestrelV2DockInfoHeader {
+		guint8 total_devices;
+		guint8 first_index;
+		guint8 last_index;
+	} header;
+	struct FuDellKestrelEcQueryEntry {
+		struct FuDellKestrelV2EcAddrMap {
+			guint8 location;
+			guint8 device_type;
+			guint8 sub_type;
+			guint8 arg;
+			guint8 instance;
+		} ec_addr_map;
+		union {
+			guint32 version_32;
+			guint8 version_8[4];
+		} __attribute__((packed)) version; /* nocheck:blocked */
+	} devices[20];
+} FuDellKestrelDockInfoStructure;
```

Comment:
can you please convert this to rs?  We shouldn't have nocheck stuff added to the codebase now.

Reply:
it seems rust cannot convert the array of struct very well, I've tried below but the size the rust code given to it only `12` bytes i.e.: `hdr (3 bytes)` + `EcQueryEntry (9 bytes)`.  Supposedly it should be `3` + `9 * 20`.

```
#[repr(C, packed)]
#[derive(Getters, ToString)]
struct FuStructDellKestrelDockInfoEcQueryEntry {
    ec_addr_map: FuStructDellKestrelDockInfoEcAddrMap,
    version_32: u32be,
}

#[repr(C, packed)]
#[derive(New, Getters, Parse, ToString)]
struct FuStructDellKestrelDockInfo {
    header: FuStructDellKestrelDockInfoHeader,
    devices: [FuStructDellKestrelDockInfoEcQueryEntry; 20],
}
```
Below is the rust code:
```
/**
 * fu_struct_dell_kestrel_dock_info_new: (skip):
 **/
FuStructDellKestrelDockInfo *
fu_struct_dell_kestrel_dock_info_new(void)
{
    FuStructDellKestrelDockInfo *st = g_byte_array_sized_new(12);
    fu_byte_array_set_size(st, 12, 0x0);
    {
        g_autoptr(GByteArray) st_donor = fu_struct_dell_kestrel_dock_info_header_new();
        memcpy(st->data + 0x0, st_donor->data, st_donor->len); /* nocheck:blocked */
    }
    {
        g_autoptr(GByteArray) st_donor = fu_struct_dell_kestrel_dock_info_ec_query_entry_new();
        memcpy(st->data + 0x3, st_donor->data, st_donor->len); /* nocheck:blocked */
    }
    return st;
}
```

Perhaps we can optimize it later, and move this plugin to 1.9.27 first?

Reply:
@CragW can you file a bug with a tiny reproducer in `libfwupdplugin/fu-self-test.rs` and I'll fix tomorrow?

Reply:
Created #8096

Reply:
Can you rebase on main and try again now?